demos/build_demos.sh

```diff
@@ -26,12 +26,18 @@ error() {
 trap 'error ${LINENO}' ERR
 
 extra_cmake_opts=()
+build_targets=()
 
 for opt in "$@"; do
     case "$opt" in
     -DENABLE_PYTHON=*)
         extra_cmake_opts+=("$opt")
         ;;
+    --target=*)
+        tmp="${opt%\"}"
+        tmp="${tmp#\"}"
+        build_targets+=("${tmp//=/ }")
```

Comment:
I'm confused by this. AFAICS, this replaces all equals signs in the option with spaces and than adds that option to `build_targets`. But doesn't the option still contain the string `--target` at that point? Why would you add _that_ to `build_targets`?

Reply:
Also, is it really worth it to support combining multiple `--target` options? CMake itself doesn't support it, so why should we?

Reply:
Fixed.  
About the neccesity of such option, if it's hard to support or test we can remove it.

Reply:
It's not about the difficulty of support (I don't think it's that difficult). I just think that if we implement the support for multiple `--target` options now, we might end up not matching CMake if CMake developers decide to implement it with different semantics.

For example, in your current implementation, `--target=a --target=b` is equivalent to `--target 'a b'`. But in a future CMake version, it could be equivalent to `--target b` instead (i.e., they could make the last option override any previous ones). Since we're trying to mimic CMake options here, I think that's an undesirable outcome. Therefore, it would be better to reject multiple `--target` options, like CMake is doing now.

Reply:
I get your point. I will remove it.

Reply:
I'm quite confused as to what you did here. Now multiple `--target` options are _still_ supported, but multiple targets aren't? That makes no sense.

Reply:
Now, I just pass --target to cmake, without calling `cmake --build .  --target A` multiple times in cycle. If cmake version supports multiple targets, they will build, if not - there will be an error from cmake side. Why does it make no sense?

Reply:
I just let cmake decides how to interpret --target option, not to mimic it, like you suggest.

Reply:
(Sorry, it seems that this PR has slipped my mind.)

I have examined everything again, and it seems that I have made a critical mistake - CMake 3.15+ _does_, in fact, support multiple `--target` options (although I don't see this advertised in the documentation). Given this, your solution does indeed make sense.

On a separate note, it's a bit unfortunate that your syntax doesn't correspond to `cmake --build` exactly (since you require the target(s) to be passed via the same argument as the option name, whereas CMake requires them to be passed as separate arguments), but replicating CMake's parsing logic would be a pain and not really worth it.